### ISSUE TITLE
Build fails since bir cache is not created

### ISSUE THREAD

**Issue by dinuish94** [ballerina-platform/ballerina-lang#19043]
**Description:**
When an external module (A) is referenced as a dependency in module (B), the build fails to create the bir cache in the target folder of module (B). Hence throwing the below exception.

```
error: /wso2/healthcare/0.1.0/healthcare.bir not found in /Users/dinuksha/Documents/Projects/EI/ballerina-integrator/docs/learn/integration-tutorials/exposing-several-services-as-a-single-service/target/caches/bir_cache /Users/dinuksha/.ballerina/bir_cache /Library/Ballerina/ballerina-1.0.0/distributions/jballerina-1.0.0/bir-cache 
	at ballerina.compiler_backend_jvm:findCacheDirFor(jvm_package_gen.bal:330)
	   ballerina.compiler_backend_jvm:lookupModule(jvm_package_gen.bal:309)
	   ballerina.compiler_backend_jvm:generatePackage(jvm_package_gen.bal:131)
	   ballerina.compiler_backend_jvm:generateDependencyList(jvm_package_gen.bal:119)
	   ballerina.compiler_backend_jvm:generatePackage(jvm_package_gen.bal:139)
	   ballerina.compiler_backend_jvm:generateJarBinary(main.bal:72)
	   ballerina.compiler_backend_jvm:main(main.bal:47)
```

**Steps to reproduce**
This error occurs when building the exposing-several-services-as-a-single-service module which uses the healthcare-service module as a dependency in https://github.com/wso2/ballerina-integrator.

The Ballerina.toml file is of 'exposing-several-services-as-a-single-service' module is as follows.

```
[project]
org-name= "wso2"
version= "0.1.0"

[dependencies]
"wso2/healthcare" = { path = "../../backends/healthcare-service/target/balo/healthcare-2019r3-any-0.1.0.balo" }
```

**Affected Versions:**
1.0

**Comment by hemikak** [comment]
This issue occurs because we are not writing the BIRs of imports of the testable package.

**Comment by hemikak** [comment]
If we remove the "import wso2/healthcare as _;" import from tests, then it works.

**Comment by hemikak** [comment]
@Kishanthan , I have checked this issue and implemented the part that writes the BIRs of the imports in the testable package as per earlier comments with the attached diff. But once that is done I get the same error mentioned in https://github.com/ballerina-platform/ballerina-lang/issues/19558. Any idea on that ?

[testable-import-bir-write-patch.diff.txt](https://github.com/ballerina-platform/ballerina-lang/files/3800276/testable-import-bir-write-patch.diff.txt)

**Comment by suganyasuven** [comment]
This issue is already fixed with 1.0.2.
